Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -782,102 +783,187 @@ void ACesium3DTileset::MarkTilesetDirty() {
+  }
 
   return Cesium3DTiles::ViewState::create(
-      unrealWorldToTileset *
-          glm::dvec4(location.X, location.Y, location.Z, 1.0),
+      frustumType,
+      unrealWorldToTileset * glm::dvec4(
+                                 camera.location.X,
+                                 camera.location.Y,
+                                 camera.location.Z,
+                                 1.0),
       glm::normalize(
           unrealWorldToTileset *
           glm::dvec4(direction.X, direction.Y, direction.Z, 0.0)),
       glm::normalize(unrealWorldToTileset * glm::dvec4(up.X, up.Y, up.Z, 0.0)),
-      glm::dvec2(viewportSize.X, viewportSize.Y),
+      glm::dvec2(camera.viewportSize.X, camera.viewportSize.Y),
       horizontalFieldOfView,
-      verticalFieldOfView);
+      verticalFieldOfView,
+      0.01 * camera.orthographicWidth,
+      0.01 * camera.orthographicWidth / aspectRatio);
 }
 
 #if WITH_EDITOR
-std::optional<ACesium3DTileset::UnrealCameraParameters>
-ACesium3DTileset::GetEditorCamera() const {
-  FViewport* pViewport = GEditor->GetActiveViewport();
-  FViewportClient* pViewportClient = pViewport->GetClient();
-  FEditorViewportClient* pEditorViewportClient =
-      static_cast<FEditorViewportClient*>(pViewportClient);
-  const FVector& location = pEditorViewportClient->GetViewLocation();
-  const FRotator& rotation = pEditorViewportClient->GetViewRotation();
-  double fov = pEditorViewportClient->FOVAngle;
-  FVector2D size = pViewport->GetSizeXY();
-
-  if (size.X < 1.0 || size.Y < 1.0) {
-    return std::nullopt;
+std::vector<ACesium3DTileset::UnrealCameraParameters>
+ACesium3DTileset::GetEditorCameras() const {
+
+  std::vector<ACesium3DTileset::UnrealCameraParameters> cameras;
+  for (FViewportClient* pViewportClient : GEditor->GetAllViewportClients()) {
+    FEditorViewportClient* pEditorViewportClient =
+        static_cast<FEditorViewportClient*>(pViewportClient);
+
+    if (!pEditorViewportClient) {
+      continue;
+    }
+
+    const FVector& location = pEditorViewportClient->GetViewLocation();
+    double fov = pEditorViewportClient->FOVAngle;
+    FIntPoint _;
+    FIntPoint size;
+    pEditorViewportClient->GetViewportDimensions(_, size);
+
+    FRotator rotation;
+
+    switch (pEditorViewportClient->GetViewportType()) {
+    case LVT_OrthoXY:
+      rotation = FRotator(-90.0f, -90.0f, 0.0f);
+      break;
+    case LVT_OrthoNegativeXY:
+      rotation = FRotator(90.0f, 90.0f, 0.0f);
+      break;
+    case LVT_OrthoXZ:
+      rotation = FRotator(0.0f, -90.0f, 0.0f);
+      break;
+    case LVT_OrthoNegativeXZ:
+      rotation = FRotator(0.0f, 90.0f, 0.0f);
+      break;
+    case LVT_OrthoYZ:
+      rotation = FRotator(0.0f, 0.0f, 0.0f);
+      break;
+    case LVT_OrthoNegativeYZ:
+      rotation = FRotator(0.0f, 180.0f, 0.0f);
+      break;
+    case LVT_OrthoFreelook:
+    case LVT_Perspective:
+    case LVT_MAX:
+    case LVT_None:
+      rotation = pEditorViewportClient->GetViewRotation();
```

Comment:
Is there a reason why these cases are handled individually? 

E.g. when the viewport type is `LVT_OrthoXZ`, I'd expect `GetViewRotation` to exactly return `FRotator(0.0f, -90.0f, 0.0f)`. Or will `GetViewRotation` return different values than the hard-codes ones?

Reply:
I initially used GetViewRotation, but it seemed to return FRotator(0, 0, 0). At least when testing with orthographic editor viewports.